### PR TITLE
package favouritism

### PR DESCRIPTION
**Pull request by EmileTrotignon**
This add an option to choose which package are going to be favoured in the ranking.

For now, there is no way to specifiy no packages, I will change that.

### CODE REVIEW

File: cli/main.ml [hunk 1]
```diff
@@ -53,7 +53,7 @@ let cmd_search =
 let cmd_index =
   let doc = "Index odocl files to create a Sherlodoc database" in
   let info = Cmd.info "index" ~doc in
-  Cmd.v info (with_db Index.term db_filename)
+  Cmd.v info Term.(with_db Index.term db_filename)
```
Comment (art-w) [inline cli/main.ml:56]: Is this change required?

Comment (EmileTrotignon) [reply cli/main.ml:56]: No, its probably an artifact from rebase

File: index/load_doc.ml [hunk 2]
```diff
@@ -34,10 +34,14 @@ let cost_doc = function
     0
   | _ -> 100
 
-let cost ~name ~kind ~doc_html ~rhs ~cat =
+let cost ~name ~kind ~doc_html ~rhs ~cat ~favoured_packages =
   String.length name
   + (5 * path_length name)
-  + (if string_starts_with ~prefix:"Stdlib." name then 0 else 50)
+  + (if List.exists
+          (fun package -> string_starts_with ~prefix:(package ^ ".") name)
```
[the remark this answers is not in the document]
Comment (EmileTrotignon) [reply index/load_doc.ml:41]: I think user are going to be surprised by the behaviour if we don't add the dot, because its not named "favoured prefixes" but "favoured packages". Maybe we can change both ?

[the remark this answers is not in the document]
Comment (art-w) [reply index/load_doc.ml:41]: Well the module name we match against isn't a package either :P For the integration with dune, how will you fill that flag?

[the remark this answers is not in the document]
Comment (EmileTrotignon) [reply index/load_doc.ml:41]: I do not know yet, but if I do not find a good way to get a package name I will make it empty

**Comment by art-w** [comment]
Thanks! I'm not too crazy that we have different flags for specifying paths to odocl files, but I suppose the `--favoured` will only be used by dune anyway... (I hope we never hit the `getconf ARG_MAX` max length for command-line arguments, but I'm guessing we would need >40k paths of favored odocls for that to happen on my system)

Do you mind cleaning a bit the commit history so we can merge? :)

**Review by art-w — APPROVED** [review]
Nice, thanks again!
